src/sync.rs

```diff
@@ -199,12 +209,29 @@ impl LatestState {
 pub fn sync(
     stdout: &mut StandardStream,
     info_color_spec: ColorSpec,
-    mail_dir: PathBuf,
     args: Args,
     config: Config,
 ) -> Result<(), Error> {
+
+    let state_dir = match &config.state_dir {
+        Some(ref dir) => dir.clone(),
```

Comment:
This needs special handling if it's a relative path, in which case it should be relative to the root mail dir as mentioned above.

Reply:
This will always be set and absolute now up in config so we don't have to worry.